compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala

```diff
@@ -26,17 +26,14 @@ object QuoteContextImpl {
   def apply()(using Context): QuoteContext =
     new QuoteContextImpl(ctx)
 
-  def showTree(tree: tpd.Tree)(using Context): String = {
-    val qctx = QuoteContextImpl()(using MacroExpansion.context(tree))
-    val syntaxHighlight =
-      if (ctx.settings.color.value == "always") SyntaxHighlight.ANSI
-      else SyntaxHighlight.plain
-    show(using qctx)(tree.asInstanceOf[qctx.reflect.Tree], syntaxHighlight)(using ctx.asInstanceOf[qctx.reflect.Context])
+  def showDecompiledTree(tree: tpd.Tree)(using Context): String = {
+    val qctx: QuoteContextImpl = new QuoteContextImpl(MacroExpansion.context(tree))
+    if ctx.settings.color.value == "always" then
+      qctx.reflect.TreeMethodsImpl.extension_showAnsiColored(tree)
+    else
+      qctx.reflect.TreeMethodsImpl.extension_show(tree)
   }
```

Comment:
Does the name change `showTree -> showDecompiledTree` suggest that the method only works for decompiled trees?

Reply:
This method is only used internally for decompilation. I renamed it to make it clear what it should do to avoid using it somewhere else afterward and changing something inside to adapt to the new use. It also differentiates it from the other general-purpose `showTree` methods.